draft-barnes-acme.md

```diff
@@ -427,6 +435,24 @@ of {{RFC7515}}.  If the value of a "nonce" header parameter is not
 valid according to this encoding, then the verifier MUST reject the
 JWS as malformed.
 
+## Key Agreement
+
+Certain elements of the protocol will require the establishment of a shared secret between the client and the server, in such a way that an entity observing the ACME protocol cannot derive the secret.  In these cases, we use a simple ECDH key exchange, based on the system used by CMS {{RFC5753}}:
+
+* Inputs:
+  * Client-generated key pair
+  * Server-generated key pair
+  * Length of the shared secret to be derived
+* Perform the ECDH primitive operation to obtain X_P (Section 3.3.1 of {{SEC1}})
+* Select a hash algorithm according to the curve being used:
+  * For "P-256", use SHA-256
+  * For "P-384", use SHA-384
+  * For "P-512", use SHA-512
+* Derive the shared secret value using the KDF in Section 3.6.1 of {{SEC1}} using the selected hash algorithm
```

Comment:
It seems like you're really burying the lede here, since the point of this exercise is to generate
a key which is << 256 bits, but you don't explain how to do that.